weblate/trans/views/edit.py

```diff
@@ -638,12 +638,27 @@ def translate(request: AuthenticatedHttpRequest, path):
             messages.error(request, gettext("Invalid search string!"))
             return redirect(obj)
 
+    from_offset = int(request.GET.get("from_offset", offset - 1))
+    if from_offset and 1 <= from_offset <= len(search_result["ids"]):
+        previous_unit = unit_set.get(pk=search_result["ids"][from_offset - 1])
+
+        if unit.translation.component != previous_unit.translation.component:
+            messages.warning(
+                request,
+                gettext("You have shifted from %(previous)s to %(current)s.")
+                % {
+                    "previous": previous_unit.translation.component.name,
+                    "current": unit.translation.component.name,
```

Comment:
Name is not good enough here as it does not have to be unique, at least categories should be included. Perhaps using full component name including project would be better for consistency with other views.